classes/Event/class.xoctEventGUI.php

```diff
@@ -890,7 +890,16 @@ public function opencaststudio(): void
             $this->main_tpl->setOnScreenMessage('failure', $this->txt('msg_no_access'), true);
             $this->cancel();
         }
-        $this->addCurrentUserToProducers();
+        // Consider studio group as default.
+        $user_group_name = PluginConfig::F_GROUP_STUDIO;
+
+        // Looking for "Edit Video" permission, to add user to producers group.
+        if (ilObjOpenCastAccess::hasPermission(ilObjOpenCastAccess::PERMISSION_EDIT_VIDEOS)) {
```

Comment:
What happens if a person no longer has the permission to edit videos? maybe we should remove this person then?

Reply:
If the person no longer has the EDIT VIDEO perm, it falls back to RECORD perm, and if he does not have that one as well, the check permission at the beginning of the function will prevent the access.

PS: I have also mentioned this, but according to @dagraf, the current checkers (mentioned above) are enough to handle this scenario.

Reply:
A user may no longer have the permission to edit videos in series A but may still have the permission to edit videos in series B. Therefore, I think we should not remove a person that has no longer the permission to edit videos.